core/commands/cid.go

```diff
@@ -0,0 +1,314 @@
+		baseStr, _ := req.Options["b"].(string)
+
+		opts := cidFormatOpts{}
+
+		if strings.IndexByte(fmtStr, '%') == -1 {
+			resp.SetError(fmt.Errorf("invalid format string: %s", fmtStr), cmdkit.ErrNormal)
+			return
+		}
+		opts.fmtStr = fmtStr
+
+		switch verStr {
+		case "":
+			// noop
+		case "0":
+			opts.verConv = toCidV0
+		case "1":
+			opts.verConv = toCidV1
+		default:
+			resp.SetError(fmt.Errorf("invalid cid version: %s\n", verStr), cmdkit.ErrNormal)
+			return
+		}
+
+		if baseStr != "" {
+			encoder, err := mbase.EncoderByName(baseStr)
+			if err != nil {
+				resp.SetError(err, cmdkit.ErrNormal)
+				return
+			}
+			opts.newBase = encoder.Encoding()
+		} else {
+			opts.newBase = mbase.Encoding(-1)
+		}
+
+		emitCids(req, resp, opts)
+	},
+	PostRun: cmds.PostRunMap{
+		cmds.CLI: streamRes(func(v interface{}, out io.Writer) nonFatalError {
+			r := v.(*CidFormatRes)
+			if r.ErrorMsg != "" {
+				return nonFatalError(fmt.Sprintf("%s: %s", r.CidStr, r.ErrorMsg))
+			}
+			fmt.Fprintf(out, "%s\n", r.Formatted)
+			return ""
+		}),
+	},
+	Type: CidFormatRes{},
+}
+
+type CidFormatRes struct {
+	CidStr    string // Original Cid String passed in
+	Formatted string // Formated Result
+	ErrorMsg  string // Error
+}
+
+var base32Cmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Convert CIDs to Base32 CID version 1.",
+	},
+	Arguments: []cmdkit.Argument{
+		cmdkit.StringArg("cid", true, true, "Cids to convert.").EnableStdin(),
+	},
+	Run: func(req *cmds.Request, resp cmds.ResponseEmitter, env cmds.Environment) {
+		opts := cidFormatOpts{
+			fmtStr:  "%s",
+			newBase: mbase.Encoding(mbase.Base32),
+			verConv: toCidV1,
+		}
+		emitCids(req, resp, opts)
+	},
+	PostRun: cidFmtCmd.PostRun,
+	Type:    cidFmtCmd.Type,
+}
+
+type cidFormatOpts struct {
+	fmtStr  string
+	newBase mbase.Encoding
+	verConv func(cid *cid.Cid) (*cid.Cid, error)
+}
+
+func emitCids(req *cmds.Request, resp cmds.ResponseEmitter, opts cidFormatOpts) {
+	out := make(chan interface{}, 128)
+	ctx := req.Context
+	go func() {
+		defer close(out)
+		for _, cidStr := range req.Arguments {
+			emit := func(fmtd string, err error) {
+				res := &CidFormatRes{CidStr: cidStr, Formatted: fmtd}
+				if err != nil {
+					res.ErrorMsg = err.Error()
+				}
+				select {
+				case <-ctx.Done():
+					resp.SetError(ctx.Err(), cmdkit.ErrNormal)
+				case out <- res:
+				}
+			}
+			c, err := cid.Decode(cidStr)
+			if err != nil {
+				emit("", err)
+				continue
+			}
+			base := opts.newBase
+			if base == -1 {
+				base, _ = cid.ExtractEncoding(cidStr)
+			}
+			if opts.verConv != nil {
+				c, err = opts.verConv(c)
+				if err != nil {
+					emit("", err)
+					continue
+				}
+			}
+			str, err := cidutil.Format(opts.fmtStr, base, c)
+			if _, ok := err.(cidutil.FormatStringError); ok {
+				// no point in continuing if there is a problem with the format string
+				resp.SetError(err, cmdkit.ErrNormal)
+				return
+			}
+			emit(str, err)
+		}
+	}()
+	resp.Emit(out)
```

Comment:
You can just call Emit on each CID individually (no need for the goroutine or the channel).